code/modules/martial_arts/martial.dm

```diff
@@ -9,6 +9,9 @@
 	var/no_guns = FALSE	//set to TRUE to prevent users of this style from using guns
 	var/no_guns_message = ""	//message to tell the style user if they try and use a gun while no_guns = TRUE (DISHONORABRU!)
 	var/temporary = 0
+	var/weapon_affinity	//if this martial art has any interaction with a weapon, also spawns said weapon when the manual is used
```

Comment:
I would advice to add a used variable to the martial art book to avoid it being used to spawn in a unlimited number of weapons

Reply:
The books are deleted when you use it once.